python/pyiceberg/table/__init__.py

```diff
@@ -228,12 +225,9 @@ def select(self, *field_names: str) -> S:
             return self.update(selected_fields=field_names)
         return self.update(selected_fields=tuple(set(self.selected_fields).intersection(set(field_names))))
 
-    def filter_rows(self, new_row_filter: BooleanExpression) -> S:
+    def filter(self, new_row_filter: BooleanExpression) -> S:
```

Comment:
I couldn't really decide on a name here since `filter` is an internal keyword. But couldn't think of anything better